kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/resolver/ResolutionResult.java

```diff
@@ -36,18 +39,84 @@ public class ResolutionResult {
     private final Map<LocalRef<KafkaService>, KafkaService> kafkaServiceRefs;
     private final Set<ClusterResolutionResult> clusterResolutionResults;
 
-    public record ClusterResolutionResult(VirtualKafkaCluster cluster, Set<LocalRef<?>> unresolvedDependencySet) {
+    public Optional<ClusterResolutionResult> clusterResult(VirtualKafkaCluster cluster) {
+        return clusterResolutionResults.stream()
+                .filter(r -> r.cluster == cluster)
+                .findFirst();
+    }
+
+    enum DependencyType {
+        // from the virtual kafka cluster to another entity
+        DIRECT,
+        // from any other entity to another entity
+        TRANSITIVE
+    }
+
+    public record UnresolvedReferences(Set<UnresolvedReference> unresolved,
+                                       Set<KafkaProxyIngress> ingressesWithResolvedRefsFalse,
+                                       Set<KafkaProtocolFilter> filtersWithResolvedRefsFalse,
+                                       Set<KafkaService> kafkaServicesWithResolvedRefsFalse) {
+        public UnresolvedReferences {
+            Objects.requireNonNull(unresolved);
+        }
+
+        public Stream<LocalRef<?>> getUnresolvedReferences(LocalRef<?> from, String kindTo) {
+            Objects.requireNonNull(from);
+            Objects.requireNonNull(kindTo);
+            return unresolved.stream().filter(r -> r.from.equals(from) && r.to.getKind().equals(kindTo)).map(UnresolvedReference::to);
+        }
+
+        public Stream<LocalRef<?>> getUnresolvedReferences(String fromKind, String toKind) {
+            Objects.requireNonNull(fromKind);
+            Objects.requireNonNull(toKind);
+            return unresolved.stream().filter(r -> r.from.getKind().equals(fromKind) && r.to.getKind().equals(toKind)).map(UnresolvedReference::to);
+        }
+
+        public boolean hasUnresolvedReferences(String fromKind, String toKind) {
```

Comment:
I question whether you really need this method.

Reply:
removed